### PR TITLE
refactor: Move RobustMultiNewton to NonlinearSolveFirstOrder

### PR DESCRIPTION
**Pull request by ErikQQY**
Continue #494

### CODE REVIEW

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Test failure looks unrelated.

**Comment by avik-pal** [comment]
No it is related, it needs an implicit import

File: lib/NonlinearSolveFirstOrder/src/poly_algs.jl [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        algs = (
+            TrustRegion(; common_kwargs...),
+            TrustRegion(; common_kwargs..., radius_update_scheme = RUS.Bastin),
+            NewtonRaphson(; common_kwargs...),
+            NewtonRaphson(; common_kwargs..., linesearch = BackTracking()),
+            TrustRegion(; common_kwargs..., radius_update_scheme = RUS.NLsolve),
+            TrustRegion(; common_kwargs..., radius_update_scheme = RUS.Fan)
+        )
+    end
+    return NonlinearSolvePolyAlgorithm(algs)
```
Comment (avik-pal) [inline lib/NonlinearSolveFirstOrder/src/poly_algs.jl:43]: change to `NonlinearSolveBase.` or add `using NonlinearSolveBase: NonlinearSolvePolyAlgorithm`

Comment (ErikQQY) [reply lib/NonlinearSolveFirstOrder/src/poly_algs.jl:43]: Oh yeah, just fixed

**Comment by avik-pal** [comment]
The MTK tests are unrelated. @AayushSabharwal the MTK tests seem incorrect, no-init cache doesn't have `u`

**Comment by AayushSabharwal** [comment]
NonlinearSolve caches don't follow their own interface 😅 I'll PR to fix this

**Comment by AayushSabharwal** [comment]
https://github.com/SciML/NonlinearSolve.jl/pull/506

**Pull request merged**